espnet/nets/pytorch_backend/encoders.py

```diff
@@ -100,25 +105,37 @@ def __init__(self, idim, elayers, cdim, hdim, dropout, typ="blstm"):
             self.l_last = torch.nn.Linear(cdim, hdim)
         self.typ = typ
 
-    def forward(self, xs_pad, ilens):
+    def forward(self, xs_pad, ilens, prev_state=None):
         """RNN forward
 
         :param torch.Tensor xs_pad: batch of padded input sequences (B, Tmax, D)
         :param torch.Tensor ilens: batch of lengths of input sequences (B)
+        :param torch.Tensor prev_state: batch of previous RNN states
         :return: batch of hidden state sequences (B, Tmax, eprojs)
         :rtype: torch.Tensor
         """
         logging.info(self.__class__.__name__ + ' input lengths: ' + str(ilens))
         xs_pack = pack_padded_sequence(xs_pad, ilens, batch_first=True)
         self.nbrnn.flatten_parameters()
-        ys, _ = self.nbrnn(xs_pack)
+        if prev_state is not None and self.nbrnn.bidirectional:
+            prev_state = zero_backward_rnn_state(prev_state)
+        ys, states = self.nbrnn(xs_pack, hx=prev_state)
         # ys: utt list of frame x cdim x 2 (2: means bidirectional)
         ys_pad, ilens = pad_packed_sequence(ys, batch_first=True)
         # (sum _utt frame_utt) x dim
         projected = torch.tanh(self.l_last(
             ys_pad.contiguous().view(-1, ys_pad.size(2))))
         xs_pad = projected.view(ys_pad.size(0), ys_pad.size(1), -1)
-        return xs_pad, ilens  # x: utt list of frame x dim
+        return xs_pad, ilens, states  # x: utt list of frame x dim
+
+
+def zero_backward_rnn_state(states):
```

Comment:
It's better to comment about the role of this function.

Reply:
:+1: